### ISSUE TITLE
i build icmp-redirect.c in 'sample' dir.and seems not work.

### ISSUE THREAD

**Issue by RyanCode** [libnet/libnet#151]
```shell
./icmp-redirect -s 192.168.1.10 -d 192.168.1.9 -g 192.168.1.1                    1 ↵  
libnet 1.1 packet shaping: ICMP redirect[link]
Write error: libnet_write_link(): -1 bytes written (Device not configured)
```
which device i didn't configure?

**Comment by troglobit** [comment]
Testing latest git master:
```
$ sudo ./icmp_redirect -s 192.168.1.10 -d 192.168.1.9 -g 192.168.1.1 
libnet 1.1 packet shaping: ICMP redirect[link]
Wrote 70 byte ICMP packet; check the wire.
```

tcpdump in other window:
```
$ tcpdump -i eth0 -vvv icmp
tcpdump: listening on eth0, link-type EN10MB (Ethernet), snapshot length 262144 bytes
10:40:02.175169 IP (tos 0x18, ttl 64, id 238, offset 0, flags [none], proto ICMP (1), length 56)
    192.168.1.10 > 192.168.1.9: ICMP redirect 192.168.1.10 to host OpenWrt.lan, length 36
	IP (tos 0x18, ttl 64, id 66, offset 0, flags [none], proto ICMP (1), length 28)
^C
1 packet captured
1 packet received by filter
0 packets dropped by kernel
```

Cannot reproduce, can you provide some more details:
 - what operating system are you using
 - which version of libnet (git master?)

**Comment by RyanCode** [comment]
> 
Macos,13-inch, M1
I installed libnet-1.2 from brew.
tcpdump -i en0 -vvv icmp does works.

**Comment by troglobit** [comment]
Ah, sorry, libnet is *very* untested on macOS.  So unfortunately you're on your own here to debug.

**Comment by RyanCode** [comment]
> Ah, sorry, libnet is _very_ untested on macOS. So unfortunately you're on your own here to debug.

alright,thanks

**Comment by troglobit** [comment]
I decided to dig up my old MacBook Pro and test this over the holidays.

**Comment by troglobit** [comment]
Turns out we had a major regression introduced quite recently that now has been fixed, as of commit f7f0a11.  I hope this is helpful to you.
